### PR TITLE
[Enhancement] Show history records in proc dir /compactions

### PR DESCRIPTION
**Pull request by sduzh**
```
> show proc "/compactions";
+---------------+-------+---------------------+---------------------+---------------------+------------------------------------------------------------------------+
| Partition     | TxnID | StartTime           | CommitTime          | FinishTime          | Error                                                                  |
+---------------+-------+---------------------+---------------------+---------------------+------------------------------------------------------------------------+
| zhuming.t0.t0 | 3319  | 2023-04-20 12:50:58 | 2023-04-20 12:50:59 | 2023-04-20 12:50:59 | NULL                                                                   |
| zhuming.t0.t0 | 3315  | 2023-04-20 12:36:14 | NULL                | 2023-04-20 12:36:14 | A error occurred: errorCode=2001 errorMessage:Connection reset by peer |
| zhuming.t0.t0 | 3316  | 2023-04-20 12:36:20 | NULL                | 2023-04-20 12:36:20 | Unable to validate object                                              |
| zhuming.t0.t0 | 3317  | 2023-04-20 12:36:26 | NULL                | 2023-04-20 12:36:26 | Unable to validate object                                              |
| zhuming.t0.t0 | 3318  | 2023-04-20 12:50:51 | NULL                | 2023-04-20 12:50:52 | starlet err server address is still unknown                            |
+---------------+-------+---------------------+---------------------+---------------------+------------------------------------------------------------------------+
```
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 16 / 88 (18.18%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/lake/compaction/CompactionManager.java|0|1|00.00%|[167]|
|:large_blue_circle:|com/starrocks/lake/compaction/CompactionContext.java|0|6|00.00%|[38, 39, 41, 42, 98, 102]|
|:large_blue_circle:|com/starrocks/lake/compaction/CompactionRecord.java|0|17|00.00%|[28, 29, 30, 31, 32, 33, 34, 35, 36, 39, 43, 47, 51, 55, 59, 63, 67]|
|:large_blue_circle:|com/starrocks/common/proc/CompactionsProcNode.java|1|10|10.00%|[38, 40, 41, 43, 44, 45, 46, 47, 48]|
|:large_blue_circle:|com/starrocks/lake/compaction/CompactionScheduler.java|13|52|25.00%|[141, 142, 155, 156, 159, 277, 278, 290, 291, 356, 357, 358, 360, 421, 422, 423, 424, 425, 426, 427, 438, 439, 444, 445, 454, 455, 456, 457, 458, 459, 460, 463, 464, 465, 466, 469, 470, 471, 472]|
|:large_blue_circle:|com/starrocks/common/Config.java|2|2|100.00%|[]|

**Review by tracymacding — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/lake/compaction/CompactionScheduler.java [hunk 1]
```diff
@@ -402,7 +417,58 @@ private void abortTransactionIgnoreError(long dbId, long txnId, String reason) {
...
+    }
+
+    private static class SynchronizedCircularQueue<E> {
+        private CircularFifoQueue<E> q;
+
+        SynchronizedCircularQueue(int size) {
+            q = new CircularFifoQueue<>(size);
+        }
+
+        int size() {
```
Comment (kevincai) [inline fe/fe-core/src/main/java/com/starrocks/lake/compaction/CompactionScheduler.java:437]: synchronized int size() { ... }

**Review by kevincai — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/lake/compaction/CompactionScheduler.java (hunk 1 above)
Comment (sduzh) [reply fe/fe-core/src/main/java/com/starrocks/lake/compaction/CompactionScheduler.java]: Updated, PTAL

**Review by kevincai — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 15 / 78 (19.23%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/lake/compaction/CompactionManager.java|0|1|00.00%|[167]|
|:large_blue_circle:|com/starrocks/lake/compaction/CompactionContext.java|0|6|00.00%|[38, 39, 41, 42, 98, 102]|
|:large_blue_circle:|com/starrocks/lake/compaction/CompactionRecord.java|0|17|00.00%|[28, 29, 30, 31, 32, 33, 34, 35, 36, 39, 43, 47, 51, 55, 59, 63, 67]|
|:large_blue_circle:|com/starrocks/common/proc/CompactionsProcNode.java|1|10|10.00%|[38, 40, 41, 43, 44, 45, 46, 47, 48]|
|:large_blue_circle:|com/starrocks/lake/compaction/CompactionScheduler.java|12|42|28.57%|[141, 142, 155, 156, 159, 277, 278, 290, 291, 356, 357, 358, 360, 421, 422, 423, 424, 425, 426, 427, 441, 442, 443, 444, 445, 446, 449, 450, 453, 454]|
|:large_blue_circle:|com/starrocks/common/Config.java|2|2|100.00%|[]|

**Review by tracymacding — APPROVED** [review]

**Review by wyb — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Pull request merged**